physics/cires_vert_orodis_v1.F90

```diff
@@ -0,0 +1,1026 @@
+             wfdT    = wfiT/Cdft  *BetaT
+             mkzi    = 2.0*mkzm*(wfdM+wfdT)
+
+             fdis    = tauz(k-1)*exp(-mkzi*(zpi(k)-zpi(k-1)) )
+             tauz(k) = fdis
+             hsat2   = fcrit_sm2 * uf2 *bn2i(k)
+             fsat    = rhoi(k)* hsat2 * sqrt(uf2) * bvi(k)
+             if (fdis > fsat) then 
+               tauz(k)  = min(fsat, tauz(k-1))
+!=================================================================
+! two definitions for eddy mixing of MF:
+!  a) wave damping-Lindzen :  Ked ~ kx/(2H)*(u-c)^4/N^3
+!  b) heat-based turbulence: 4/3 Richardson Ked ~eps^1/3 *Lt^4/3 
+!=================================================================
+               kds     = rhp2*kxs*uf2*uf2/bn2i(k)/bvi(k)
+               kdis(k) = kds 
+             endif         
+           else
+             tauz(k:levs) = 0.     ! wave is reflected above
+             kds  = 0.
+           endif  
+         enddo
+
+         do k=krefj+1, kcrit
+           drtau(k) = rgrav*(tauz(k+1)-tauz(k))/dp(k)
+         enddo         
+!
+!
+      end  subroutine ugwp_oro_lsatdis
+!
+!
+     subroutine ugwp_tofd(im, levs, sigflt, elvmax, zpbl,  u, v, zmid, &
+                          utofd, vtofd, epstofd, krf_tofd)
+       use machine ,      only : kind_phys 
+       use ugwp_common_v1  , only :  rcpd2         
+       use ugwp_oro_init_v1, only : n_tofd, const_tofd, ze_tofd, a12_tofd, ztop_tofd
+!       
+     implicit none
+!     
+     integer ::  im, levs
+     real(kind_phys), dimension(im, levs)  ::  u, v, zmid
+     real(kind_phys), dimension(im)        ::  sigflt, elvmax, zpbl
+     real(kind_phys), dimension(im, levs)  ::  utofd, vtofd, epstofd, krf_tofd
+!
+! locals
+!
+     integer :: i, k
+     real    :: sgh = 30.
+     real    :: sgh2, ekin, zdec, rzdec, umag, zmet, zarg, zexp, krf
+!     
+     utofd =0.0 ; vtofd = 0.0 ;  epstofd =0.0 ; krf_tofd =0.0    
+!      
+     
+     do i=1, im 
+       
+       zdec = max(n_tofd*sigflt(i), zpbl(i))
+       zdec = min(ze_tofd, zdec)
+       rzdec = 1.0/zdec
+       sgh2  = max(sigflt(i)*sigflt(i), sgh*sgh)
+       
+      do k=1, levs
+         zmet = zmid(i,k)   
+      if (zmet > ztop_tofd) cycle
+       ekin = u(i,k)*u(i,k) + v(i,k)*v(i,k)
+       umag = sqrt(ekin)
+       zarg = zmet*rzdec
+       zexp = exp(-zarg*sqrt(zarg))
+       krf         = const_tofd* a12_tofd *sgh2* zmet ** (-1.2) *zexp
+       utofd(i,k)  = -krf*u(i,k)
+       vtofd(i,k)  = -krf*v(i,k)
+       epstofd(i,k)= rcpd2*krf*ekin  ! more accurate heat/mom form using "implicit tend-solver"
+                                     ! to update momentum and temp-re
+       krf_tofd(i,k) = krf
+      enddo
+     enddo
+!                
+     end subroutine ugwp_tofd  
+!
+!     
+     subroutine ugwp_tofd1d(levs, sigflt, elvmax, zsurf, zpbl,  u, v, &
+                            zmid, utofd, vtofd, epstofd, krf_tofd)
+       use machine ,      only : kind_phys 
+       use ugwp_common_v1  , only :  rcpd2         
```

Comment:
Constants should come from the host model through the argument list, if possible.

Reply:
Done.